src/Symfony/Component/Messenger/Bridge/Doctrine/Transport/Connection.php

```diff
@@ -177,7 +177,14 @@ public function get(): ?array
 
             // Append pessimistic write lock to FROM clause if db platform supports it
             $sql = $query->getSQL();
-            if (preg_match('/FROM (.+) WHERE/', (string) $sql, $matches)) {
+
+            if (method_exists(QueryBuilder::class, 'forUpdate')) {
+                $query->forUpdate();
```

Comment:
For the case of the oracle platform, this is now applying the lock on the inner query and not the outer one (see the specific code below). Is this expected ?

Reply:
it's probably better not to change the outcome here, updated